### PR TITLE
Move onShareExperience callback from party event

### PR DESCRIPTION
**Pull request by ramon-bernardo**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- Move onShareExperience callback from party event.
- Add party folder with default file for onShareExperience event callback.

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: data/scripts/eventcallbacks/party/default_onShareExperience.lua [hunk 1]
```diff
@@ -0,0 +1,29 @@
+local ec = EventCallback
+
+ec.onShareExperience = function(self, exp)
+	local sharedExperienceMultiplier = 1.20 --20%
+	local vocationsIds = {}
+	local rawExp = exp
+
+	local vocationId = self:getLeader():getVocation():getBase():getId()
+	if vocationId ~= VOCATION_NONE then
+	table.insert(vocationsIds, vocationId)
```
Comment (EPuncker) [inline data/scripts/eventcallbacks/party/default_onShareExperience.lua:10]:
```suggestion
		table.insert(vocationsIds, vocationId)
```

Comment (EPuncker) [reply data/scripts/eventcallbacks/party/default_onShareExperience.lua:10]: same with others

File: data/scripts/eventcallbacks/party/default_onShareExperience.lua [hunk 2]
```diff
@@ -0,0 +1,29 @@
+local ec = EventCallback
+
+ec.onShareExperience = function(self, exp)
```
Comment (ramon-bernardo) [inline data/scripts/eventcallbacks/party/default_onShareExperience.lua:3]: or `function(self, exp, rawExp)`?

Comment (MillhioreBT) [reply data/scripts/eventcallbacks/party/default_onShareExperience.lua:3]: rawExp is a control variable that is used to know the real experience before any call in the eventcallback system, can be ignored if you don't need it ;)

**Review by MillhioreBT — APPROVED** [review]
LGTM, including all event events in eventcallback is essential to be able to have full control of these events from revscript.

**Comment by DSpeichert** [comment]
Again - what's the point of reorganizing the datapack to move existing metatable scripts to "callbacks"?

If we had a plan to move everything and deprecate old system or standardize on one then it would make sense but it feels like those changes are suggested randomly and are unnecessary. This datapack (default one) showcases all features, not just the latest shiny callback.

Unless there's some change in behavior that I'm not seeing here, then please tell me.

**Comment by MillhioreBT** [comment]
Don, You are absolutely right, there is no urgent need to move all systems to eventcallbacks, the only real function that can be taken advantage of by moving certain things to eventcallbacks is that when we create a new eventcallbacks script of the same type we can register the event before the default and control what will happen much earlier

However this is something the user can do personally on their own custom server and the tfs master doesn't need it at all

closing PR is not bad, merge it either

**Comment by ramon-bernardo** [comment]
Is it time to merge it now? I can open another PR in a few days with this.

:triangular_flag_on_post: I'll map it for the future.

**Pull request closed without merging**